src/pdn/src/shape.cpp

```diff
@@ -617,6 +617,23 @@ Shape* Shape::extendTo(
   return new_shape.release();
 }
 
+ShapeTreeMap Shape::convertVectorToTree(const ShapeVectorMap& vec)
```

Comment:
Would it be better to clear/shrink the vec here rather than in each caller?

Reply:
done